pkg/controller/podgc/gc_controller.go

```diff
@@ -59,14 +61,18 @@ func NewPodGC(kubeClient clientset.Interface, rpClients map[string]clientset.Int
 	}
 	gcc := &PodGCController{
 		kubeClient:              kubeClient,
-		resourceProviderClients: rpClients,
 		terminatedPodThreshold:  terminatedPodThreshold,
 		deletePod: func(tenant, namespace, name string) error {
 			klog.Infof("PodGC is force deleting Pod: %v/%v/%v", tenant, namespace, name)
 			return kubeClient.CoreV1().PodsWithMultiTenancy(namespace, tenant).Delete(name, metav1.NewDeleteOptions(0))
 		},
 	}
 
+	gcc.kubeClientForNodes = map[string]clientset.Interface{"tpself": kubeClient}
```

Comment:
For scale up case, there will be duplicated kubeclients

Reply:
good point; looked at code, it seems "0" key is the special case for TP client, which could help to avoid double tp clients.